Enterprise-CMCS/managed-care-review
# Upgrade storybook to v8 and use with vite

## Summary
Upgrade storybook and adjust to work with new vite build

Changes included
- Adjust `@storybook/*` deps to version 8+ and fix breaking changes with types 
- Adding `"@storybook/react-vite"` and removing both CRA and webpack storybook plugins
- Adjusted the main storybook config file to use vite setup, also changed this file to typescript
- Addressed type warning with vite config by making this config be set up in function
- Fix dart sass 2 warnings that show up
- Group storybook dep updates together
- Delete Calendar related storybook files that did not follow proper format (those storybook files were copypasta anyway, we were temporarily adding CalendarDate to our application, may be able to remove this soon and rely on direct dependency again, but felt out of scope for this PR)
- Recentering the step indicator (quick design change)

#### Related issues


## QA guidance
Run `yarn storybook` and `yarn storybook-build` in app-web and see these commands working (except for apollo errors, which are being addresses on #2475 )
